openedx/features/caliper_tracking/transformers/problem_transformers.py

```diff
@@ -47,3 +47,40 @@ def problem_show(current_event, caliper_event):
     })
 
     return caliper_event
+
+
+def problem_save(current_event, caliper_event):
+    """
+    The browser emits problem_save events when the answer to a
+    problem is save instead of submitting; that is,
+    the user selected Save Answer.
+
+    :param current_event: default event log generated.
+    :param caliper_event: caliper_event log having some basic attributes.
+    :return: updated caliper_event.
+    """
+    caliper_object = {
+        'id': current_event['referer'],
+        'extensions': {
+            'event': current_event['event']
+        },
+        'type': 'Assessment'
+    }
+    caliper_event.update({
+        'action': 'Paused',
+        'type': 'AssessmentEvent',
+        'object': caliper_object
+    })
+    caliper_event['referrer'].update({
+        'type': 'WebPage'
+    })
+    caliper_event['actor'].update({
+        'name': current_event['username'],
+        'type': 'Person'
+    })
+    caliper_event['extensions']['extra_fields'].update({
+        'course_id': current_event['context']['course_id'],
+        'ip': current_event['ip'],
+        'user_id': current_event['context']['user_id']
```

Comment:
Why are you adding `user_id` here ? I think this is done in `base_transformer`

Reply:
Resolved